### PR TITLE
FOR-1051: Fixed filtering for timestamps.

### PR DESCRIPTION
**Pull request by YuryRybakSoftensity**

### CODE REVIEW

**Comment by YuryRybakSoftensity** [comment]
I can also add tests if needed.

File: Resource.js [hunk 1]
```diff
@@ -280,9 +280,16 @@ module.exports = function(app, route, modelName, model) {
...
       var getValue = function(value, param) {
+        var parsedValue = parseInt(value, 10);
+
         if (param.instance === 'Number') {
-          return parseInt(value, 10)
+          return parsedValue;
+        }
+
+        if (parsedValue) {
+          return new Date(parsedValue);
```
Comment (travist) [inline Resource.js]: Won't this end up converting ALL number parameters into a Date object?

**Comment by travist** [comment]
Please add tests.

File: Resource.js (hunk 1 above)
Comment (YuryRybakSoftensity) [reply Resource.js]: I added instance check, please take a look.

File: Resource.js [hunk 2]
```diff
@@ -282,6 +285,11 @@ class Resource {
       ) {
         return new Date(value);
       }
+
+      if (param.instance === 'Date' && parsedValue) {
+        return new Date(parsedValue);
```
Comment (douglaselee) [inline Resource.js]: Hi @YuryRybakSoftensity,

This seems to break queries with formats like `http://localhost:3001/form/5a98653751c0700d00363e04/submission?sort=-modified&modified__lte=2018-03-03T20:54:23`.  getValue is returning on line 290 a new Date from parsedValue which just contains the year, in this case 2018.  Before change documents were being returned, now they're not.

Comment (travist) [reply Resource.js]: Thanks @douglaselee. We are looking at it right now.

Comment (douglaselee) [reply Resource.js]: Happy to help @travist.  In my case I simply appended 'Z' to the date/times so they were treated as ISO.

Comment (travist) [reply Resource.js]: Thanks. @YuryRybakSoftensity please take a look at ^

**Pull request merged**